File: Algorithms/bubble sort/BubbleSort.java
import java.util.Arrays;






/**
 * Use {@link BubbleSort#BubbleSort(int...)} to sort your {@code int} Array
 * 
 * 
 * @author Mohammad Mahdi Malmasi
 * @version 0.1.0
 */
public class BubbleSort
{
    /**
     * This method sorts your int array with bubble sort algorithms
     * 
     * @param array : array to sort
     */
    public static void bubbleSort(int...array)
    {
        // hold values for swaping
        int hold = 0;


        for (int i = 0; i < array.length; i++)
            for (int j = i+1; j < array.length; j++)
                if (array[i] > array[j])
                {
                    hold = array[i];
                    array[i] = array[j];
                    array[j] = hold;
                }
    }



    /**
     * A simple example for {@link BubbleSort#bubbleSort(int...)}
     *  
     * @param args
     */
    public static void main(String[] args) 
    {
        System.out.println("\n\n ---------------- \n");
        System.out.println("a simple test:\n");
        

        int[] array = {8, 10, 252, 0, -34, 23, -11, 43, 50, 101, 7, 33, 12};
        System.out.println("before sort: " + Arrays.toString(array));


        bubbleSort(array);
        System.out.println("after sort: " + Arrays.toString(array));


        System.out.println("\n ---------------- \n\n");
    }
}

File: Algorithms/bubble sort/bubble_sort.py





def bubble_sort(*args):
    
    # check type of the args (because type of args is tuple and it is immutable)
    if type(args) is tuple:
        array = list(args)
    else:
        array = args
        
    
    
    
    
    
    for i in range(0, len(array)):
        for j in range(i+1, len(array)):
            if array[i] > array[j]:
                array[i], array[j] = array[j], array[i]
        
       
       
    # return sorted array 
    return array
        
        






if __name__ == "__main__":

    print("\n\n ---------------- \n")
    print("a simple test:\n")
    

    array = [2.7, 21, 43, 0.5, -1, 2.5, -11, 77, 50, 101, 72, 16]
    print(f"before sort: {array}")

    array = bubble_sort(*array)
    print(f"after sort: {array}")


    print("\n ---------------- \n\n")